src/client/index.ts

```diff
@@ -44,13 +44,7 @@ export const getWalletProviderClient = (context: {
         },
       })
         .then(validateResponse)
-        .then((res) => {
-          const contentType = res.headers.get("content-type");
-          if (contentType?.includes("application/json")) {
-            return res.json();
-          }
-          return res.text();
-        }),
+        .then<string | Record<string, unknown>>(parseRawHttpResponse),
```

Comment:
This explicit type here was need to solve a TS error.

About Zod validation, we don't know the exact shape of the response here; I think validation at this level should be handled by `typed-openapi`, because it's the one that parses the OpenAPI file and knows the shape of every response.

Anyway, we usually validate the raw response in the call site with Zod.